### PR TITLE
examples: add adjtime example application

### PR DESCRIPTION
**Pull request by michallenc**
## Summary
This application demonstrates the usage of adjtime() interface.

## Impact
None, just new example application.

## Testing
Tested on samv7 based board.

### CODE REVIEW

File: examples/adjtime/adjtime_main.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ ****************************************************************************/
+
+/****************************************************************************
+ * hello_main
+ ****************************************************************************/
+
+int main(int argc, FAR char *argv[])
+{
+  struct timeval     delta;
+  struct timeval     olddelta;
```
Comment (pkarashchenko) [inline examples/adjtime/adjtime_main.c:40]:
```suggestion
  struct timeval delta;
  struct timeval olddelta;
```

File: examples/adjtime/adjtime_main.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  int sc;
+
+  /* Set delta between real time and system tick. */
+
+  delta.tv_sec = 0;
+  delta.tv_usec = 1000000;
+
+  printf("Delta time is %ld seconds and %ld micro seconds.\n",
+          delta.tv_sec, delta.tv_usec);
```
Comment (pkarashchenko) [inline examples/adjtime/adjtime_main.c:49]:
```suggestion
  printf("Delta time is %ld seconds and %ld micro seconds.\n",
         delta.tv_sec, delta.tv_usec);
```

File: examples/adjtime/adjtime_main.c [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+  sc = adjtime(&delta, &olddelta);
+  if (sc < 0)
+    {
+      printf("ERROR: adjtime() failed: %d\n", sc);
+    }
+  else
+    {
+      printf("Returned olddelta is %ld seconds and %ld micro seconds.\n",
+              olddelta.tv_sec, olddelta.tv_usec);
```
Comment (pkarashchenko) [inline examples/adjtime/adjtime_main.c:61]:
```suggestion
      printf("Returned olddelta is %ld seconds and %ld micro seconds.\n",
             olddelta.tv_sec, olddelta.tv_usec);
```

File: examples/adjtime/adjtime_main.c [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+/****************************************************************************
+ * hello_main
+ ****************************************************************************/
+
+int main(int argc, FAR char *argv[])
+{
+  struct timeval     delta;
+  struct timeval     olddelta;
+  int sc;
```
Comment (pkarashchenko) [inline examples/adjtime/adjtime_main.c:41]: why `sc` and not `ret` as it is more widely used?

File: examples/adjtime/Makefile [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+include $(APPDIR)/Make.defs
+
+# Hello, World! built-in application info
+
+PROGNAME  = $(CONFIG_EXAMPLES_ADJTIME_PROGNAME)
+PRIORITY  = $(CONFIG_EXAMPLES_ADJTIME_PRIORITY)
+STACKSIZE = $(CONFIG_EXAMPLES_ADJTIME_STACKSIZE)
+MODULE    = $(CONFIG_EXAMPLES_ADJTIME)
+
+# Hello, World! Example
```
Comment (simbit18) [inline examples/adjtime/Makefile:30]: # Hello, World!  -> # Adjtime

File: examples/adjtime/adjtime_main.c [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+#include <nuttx/config.h>
+#include <stdio.h>
+#include <sys/time.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * hello_main
```
Comment (simbit18) [inline examples/adjtime/adjtime_main.c:34]: hello_main -> adjtime_main

**Comment by michallenc** [comment]
Thank you all for the suggestions, I will go throught this by the end of the week.

**Review by jerpelea — APPROVED** [review]

**Review by simbit18 — APPROVED** [review]

File: examples/adjtime/adjtime_main.c [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+  *value = strtol(string, NULL, 10);
+  return ret;
+}
+
+/****************************************************************************
+ * Name: parse_args
+ ****************************************************************************/
+
+static void parse_args(struct timeval *delta, int argc,
+                       char **argv)
```
Comment (pkarashchenko) [inline examples/adjtime/adjtime_main.c:85]:
```suggestion
static void parse_args(FAR struct timeval *delta, int argc,
                       FAR char **argv)
```

**Review by pkarashchenko — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

File: examples/adjtime/adjtime_main.c [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+            nargs = arg_decimal(&argv[index], &value);
+
+            delta->tv_sec = (uint32_t)value;
+            index += nargs;
+            break;
+
+          case 'u':
+            nargs = arg_decimal(&argv[index], &value);
+
+            delta->tv_usec = (uint32_t)value;
```
[the remark this answers is not in the document]
Comment (michallenc) [reply examples/adjtime/adjtime_main.c:113]: I dont think usec should be unsigned. The timeval structure is defined as

```C
struct timeval
{
  time_t tv_sec;         /* Seconds */
  long tv_usec;          /* Microseconds */
};
```

**Comment by michallenc** [comment]
@xiaoxiang781216 fixed.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**